core/src/main/kotlin/org/evomaster/core/problem/rest/classifier/probabilistic/AbstractProbabilistic400EndpointModel.kt

```diff
@@ -0,0 +1,89 @@
+package org.evomaster.core.problem.rest.classifier.probabilistic
+
+import org.evomaster.core.EMConfig
+import org.evomaster.core.problem.rest.classifier.AIModel
+import org.evomaster.core.problem.rest.classifier.ModelAccuracy
+import org.evomaster.core.problem.rest.classifier.ModelAccuracyFullHistory
+import org.evomaster.core.problem.rest.classifier.ModelAccuracyWithTimeWindow
+import org.evomaster.core.problem.rest.data.Endpoint
+import org.evomaster.core.problem.rest.data.RestCallAction
+import org.evomaster.core.search.service.Randomness
+
+/**
+ * Base class for all probabilistic classifiers working at an endpoint (400 vs. not 400).
+ *
+ * Provides:
+ * - Common properties (endpoint, warmup, encoderType, randomness, dimension, initialized flag)
+ * - Shared methods for initialization checks and accuracy estimation
+ */
+abstract class AbstractProbabilistic400EndpointModel(
+    val endpoint: Endpoint,
+    var warmup: Int = 10,
+    var dimension: Int? = null,
+    val encoderType: EMConfig.EncoderType = EMConfig.EncoderType.NORMAL,
+    val randomness: Randomness
+) : AIModel {
+
+    protected var initialized: Boolean = false
+
+    private val modelAccuracyFullHistory: ModelAccuracyFullHistory = ModelAccuracyFullHistory()
+    private val modelAccuracy: ModelAccuracy = ModelAccuracyWithTimeWindow(20)
+
+    fun getModelAccuracyFullHistory(): ModelAccuracyFullHistory = modelAccuracyFullHistory
+    fun getModelAccuracy(): ModelAccuracy = modelAccuracy
+
```

Comment:
instead of manually writing getters, could have a simple `public get` declaration below the properties declaration, eg something like:
```
private val modelAccuracy: ModelAccuracy = ModelAccuracyWithTimeWindow(20)
    public get
```
(not 100% sure of syntax in kotlin, but can check properties with private set and public get)